src/System.Data.SqlClient/src/System/Data/SqlTypes/SQLInt64.cs

```diff
@@ -211,7 +211,7 @@ public static SqlInt64 Parse(String s)
             {
                 Debug.Assert(lHigh1 == 0);
                 lPartialResult = lLow1 * lHigh2;
-                if (lPartialResult < 0 || lPartialResult > Int64.MaxValue)
+                if (lPartialResult < 0)
```

Comment:
lPartialResult is a long... it can't ever have a value greater than long.MaxValue